cmd/tools/cli/main.go

```diff
@@ -36,8 +35,5 @@ import (
 // See cadence/tools/cli/README.md for usage
 func main() {
 	app := cli.NewCliApp()
-	if err := app.Run(os.Args); err != nil {
-		_, _ = fmt.Fprintln(app.ErrWriter, err)
-		os.Exit(1)
-	}
+	_ = app.Run(os.Args) // exits on error
```

Comment:
This is not an idiomatic Go code.
I understand that you want to avoid code duplication in this case. Still, I suggest creating a wrapper function and unifying error handling there instead of making it implicit inside the app itself.
```
commoncli.ErrorHandler(app.Run(os.Args))
```

Reply:
yea, agreed it's kinda odd :\  the whole concept of "can exit(1)" is problematic for stuff like this, but it's how urfave works, and because it's mutation-heavy in its API we can't hide the `*cli.App` at all.

an external error handler may work, I can give that a try.  I think it'll play nicely with urfave's printing / return-value-ing.

Reply:
changed!  thanks for pointing it out :)